src/App.tsx

```diff
@@ -8,35 +8,97 @@ import { PostsList } from './components/PostsList';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Error } from './types/Error';
+import { Post } from './types/Post';
+import { getPosts } from './api/posts';
+import { wait } from './utils/fetchClient';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [errorMessage, setErrorMessage] = useState<Error | ''>('');
+  const [commentErrorMessage, setCommentErrorMessage] = useState<Error | ''>(
+    '',
+  );
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isSidebarVisible, setIsSidebarVisible] = useState(false);
+  const [loadingComments, setLoadingComments] = useState(false);
+  const [isCommentFormVisible, setIsCommentFormVisible] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    if (selectedUser) {
+      getPosts(selectedUser.id)
+        .then(postsFromServer => {
+          wait(1000);
+          setPosts(postsFromServer);
+        })
+        .catch(() => setErrorMessage(Error.LoadingError))
+        .finally(() => setLoading(false));
+    }
+  }, [selectedUser]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  setLoading={setLoading}
+                  setSelectedUser={setSelectedUser}
+                  selectedUser={selectedUser}
+                  setSelectedPost={setSelectedPost}
+                  setErrorMessage={setErrorMessage}
+                  setIsSidebarVisible={setIsSidebarVisible}
+                  setIsCommentFormVisible={setIsCommentFormVisible}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">No user selected</p>
-
-                <Loader />
+                {!selectedUser && !errorMessage && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {loading && !isSidebarVisible ? (
+                  <Loader />
+                ) : (
+                  <>
+                    {errorMessage && (
+                      <div
+                        className="notification is-danger"
+                        data-cy="PostsLoadingError"
+                      >
+                        {errorMessage}
+                      </div>
```

Comment:
Can we create a component for this? And what we will do if we don't have any errors after loading?

Reply:
then PostList will be shown or a message that there are no posts yet